trie.go

```
@@ -452,13 +452,160 @@ func (tr *Trie) Match(input []byte) []*Match {
+	if ambiguous := int(tr.maxLen) - 1; j > ambiguous {
+		skips += j - ambiguous
+		counted += j - ambiguous
+		if skips > dualChainSkipMax {
+			return 0, 0, false
+		}
+	}
+	if j == len(w) {
+		return skips, counted, true
+	}
+	counted += len(w) - j
+	s := uint32(rootState)
```

golangci-lint (2.12.2)

[error] unnecessary conversion

(unconvert)